File: SourceCode/SQLAPI/POS.Data/Entities/Email/EmailSMTPSetting.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace POS.Data
{
    public class EmailSMTPSetting : BaseEntity
    {
        public Guid Id { get; set; }
        [Required]
        public string Host { get; set; }
        [Required]
        public string UserName { get; set; }
        [Required]
        public string Password { get; set; }
        [Required]
        public int Port { get; set; }
        [Required]
        public bool IsDefault { get; set; }
        public string EncryptionType { get; set; }
        public string FromEmail { get; set; }
        public string FromName { get; set; }
    }
}

File: SourceCode/Angular/src/app/core/domain-classes/email-smtp-setting.ts
import { StreamInvocationMessage } from "@microsoft/signalr";
export interface EmailSMTPSetting {
  id?: string;
  host: string;
  userName: string;
  password: string;
  port: number;
  isDefault: boolean;
  encryptionType: string;
  fromEmail: string;
  fromName: string;
  toEmail?: string;
}
